posthog/api/event_definition.py

```diff
@@ -27,4 +38,52 @@ class EventDefinitionViewSet(
     search_fields = ["name"]
 
     def get_queryset(self):
+        if self.request.user.organization.is_feature_available("ingestion_taxonomy"):  # type: ignore
+            try:
+                from ee.models.event_definition import EnterpriseEventDefinition
+            except ImportError:
+                pass
+            else:
+                ee_event_definitions = EnterpriseEventDefinition.objects.raw(
+                    """
+                    SELECT *
+                    FROM ee_enterpriseeventdefinition
+                    FULL OUTER JOIN posthog_eventdefinition ON posthog_eventdefinition.id=ee_enterpriseeventdefinition.eventdefinition_ptr_id
+                    WHERE team_id = %s
+                    ORDER BY name
+                    """,
+                    params=[self.request.user.team.id],  # type: ignore
+                )
+                return ee_event_definitions
         return self.filter_queryset_by_parents_lookups(EventDefinition.objects.all()).order_by(self.ordering)
+
+    def get_object(self):
+        id = self.kwargs["id"]
+        if self.request.user.organization.is_feature_available("ingestion_taxonomy"):  # type: ignore
+            try:
+                from ee.models.event_definition import EnterpriseEventDefinition
+            except ImportError:
+                pass
+            else:
+                enterprise_event = EnterpriseEventDefinition.objects.filter(id=id).first()
```

Comment:
You could try to simplify code in these lines by doing `EnterpriseEventDefinition.objects.get_or_create(eventdefinition_ptr_id=id)`.

Or actually, why not avoid overriding this and just relying on the `get_queryset` method. You could then create the object only when a user is updating it (i.e. actually using it).

Reply:
Creating an enterprise event definition will create a copy for its parent (event definition) and since its parent already exists here we just want to downcast

For the `get_queryset` method, it returns a rawsqlquery object which is why I couldn't update/retrieve it normally and thus had to override `get_object` 😢 I had to check in with @Twixes to figure that one out